### DISCUSSION TITLE
How to force variable to exist and be a certain value

### DISCUSSION THREAD

**Discussion by heestree** [cue-lang/cue#2154] [Q&A]
Hi, I am very new to CueLang and curious how to force variable to exist with a certain value. For instance, 

In cue-question.jon
`
{
    "Setup": {
        "TestItem": true
    }
}
`
and in cue-question.cue,
`
#Def_2: {
    Setup: {
        TestItem: true
    }
}
`
when I run a command 
`
.\cue.exe vet --schema '#Def_2' cue-question.cue .\cue-question.json
`
I have no error, which is good since that is what I want. However, if I remove TestItem from json file as follows:

`
{
    "Setup": {
    }
}
`
I still get no error. What I want is cue complains that TestItem does not exist. How can I do this? That is, how can I require variable exists with specific value?

I know if I change `TestItem: true` to `TestItem: bool` in cue file, cue.exe will complain, but it will also accept `TestItem: false` which is not what I want.
Any help from you would be appreciated.

**Comment by mvdan** [comment]
@heestree it sounds like you want a required field - see https://github.com/cue-lang/cue/discussions/1951. If your CUE used a required field for TestItem, then it would need to be defined in the JSON as well.

That's a proposal we are actively working towards at the moment, so I'd suggest to take a look and subscribe :)

**Reply by heestree** [reply]
@mvdan thanks for your prompt comments. is this feature added to release version? is there any syntax example for this feature?

**Reply by mvdan** [reply]
That feature is still a proposal, so it's not yet available. You can find the syntax in the proposal itself.

**Comment by heestree** [comment]
This is my work around. If any one know better work around please let me know.

In cue-question.cue
`
#Def_1: {
Setup: {
TestItem?: false | "false" // NOTE: if exists, it should be either false or "false"
}
}

#Def_2: {
Setup: {
TestItem: true | "true" // NOTE: should exist and be either true or "true"
}
}
and in cue-question.json
{
"Setup": {
}
}
`

and .\cue.exe vet --schema '#Def_2' cue-question.cue .\cue-question.json complains with
` Setup.TestItem: incomplete value true | "true"'

This extends boolean object to union type of bool or string, but we only allow two string concretes, so this seems ok with me. At least I have complains from cue.exe when TestItem does not exist in json.

Please let me know if this work around make sense or there is a part that I am missing so that it shouldn't be used.
